camera_manager.py

```diff
@@ -0,0 +1,154 @@
+        """
+        self.encoding_params: list[int] = [cv2.IMWRITE_JPEG_QUALITY, 50]
+
+class CameraManger:
+    """
+    Camera Manager manages a list of currently connected USB cameras (at instantiation)
+    and keeps track of which cameras are being used by along with a list of their configurations (fps, etc.)
+    """
+
+    def __init__(self):
+        # List of available cameras
+        self.cameras: list[Camera] = self.__create_cameras()
+        
+    
+    def __create_cameras(self) -> list[Camera]:
+        """
+        Create a list of cameras available on the computer
+        """
+        camera_dict = get_camera_name_and_paths()
+        cameras = []
+        for camera_name, camera_path in camera_dict.items():
+            camera = Camera(camera_name, camera_path)
+            cameras.append(camera)
+        return cameras
+    
+    def __get_camera(self, camera_name: str) -> Camera:
+        """
+        Get the Camera object associated with a given camera name
+        """
+        # TODO: Raise error if camera not found
+        for camera in self.cameras:
+            if camera.name == camera_name:
+                return camera
+    
+    def get_camrea_fps(self, camera_name: str)-> int:
+        """
+        Return the fps for a camera, given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.fps
+    
+    def set_camera_fps(self, camera_name: str, fps: int):
+        """
+        Change the fps for a camera, given a camera name and fps
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        camera.fps = fps
+
+    def get_camera_encoding_params(self, camera_name: str):
+        """
+        Return the encoding parameters given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.encoding_params
+        
+    def set_camera_encoding_params(self, camera_name: str, encoding_quality: int):
+        """
+        Set the camera encoding parameters given a camera name
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        camera.encoding_params[1] = encoding_quality
+
+    def camera_is_running(self, camera_name: str) -> bool:
+        """
+        Return True if camera stream is being asked for and False if it has been ended
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        return camera.is_running
+
+    
+    def start_video_capture(self, camera_name: str) -> cv2.VideoCapture:
+        """
+        Given a camera name, return an openCV video capture object (to read frames)
+        """
+        # TODO: Raise error if camera not found
+        camera = self.__get_camera(camera_name)
+        # TODO: Check if camera is already running
```

Comment:
this is veeeery important. that backing C++ will seriously explode into a segmentation fault (or, when lucky, a runtime error) - i've had it happen 😭

Reply:
camera not found should be taken care of and for checking if the camera is running, I have it being checked further up the stack